Packs/CortexXDR/Scripts/DBotGroupXDRIncidents/DBotGroupXDRIncidents.py

```diff
@@ -0,0 +1,37 @@
+from CommonServerPython import *
+
+model_name = 'xdr_clustering'
+field_for_grouping = 'xdralerts'
+field_for_name = 'xdralerts.causalityactorprocessimagename'
+
+return_type = demisto.args()['returnWidgetType']
+
+if return_type == 'incidents':
+    res = demisto.executeCommand('DBotShowClusteringModelInfo', {
+        'searchQuery': demisto.args().get('searchQuery'),
+        'modelName': model_name,
+        'returnType': 'incidents',
+        'fieldsToDisplay': demisto.args().get('fieldsToDisplay')
+    })
+    demisto.results(res)
+elif return_type == 'summary':
+    res = demisto.executeCommand('DBotShowClusteringModelInfo', {
+        'modelName': model_name
+    })
+    demisto.results(res)
```

Comment:
With this logic if the user wants to print incidents or summary of a new trained model he will need to do it in 2 steps. First train the model using this automation without argument and then use this automation with  return_type == 'incidents' or return_type == 'summary'. You can have both functionality in a single script so it will retrain if needed and then print what the user would like (incidents, summary or json)

Reply:
The user actually does nothing - just access to the dashboard and all the scripts run in parallel on the backgroud